app/kumactl/cmd/apply/apply_test.go

```diff
@@ -179,6 +181,56 @@ var _ = Describe("kumactl apply", func() {
 		Expect(resource.Meta.GetMesh()).To(Equal(""))
 		Expect(resource.Meta.GetNamespace()).To(Equal("default"))
 	})
+	
+	It("should apply a new Dataplane resource from URL", func() {
+		// setup http server
+		go func() {
+			defer GinkgoRecover()
+			// when
+			http.Handle("/testdata/", http.StripPrefix("/testdata/", http.FileServer(http.Dir("./testdata"))))
+			err := http.ListenAndServe(port, nil)
+			// then
+			Expect(err).ToNot(HaveOccurred())
+		}()
+		
+		// given
+		rootCmd.SetArgs([]string{
+			"apply", "-f", "http://localhost:3000/testdata/apply-dataplane.yaml"},
+		)
+
+		// when
+		err := rootCmd.Execute()
+		
+		// then
+		Expect(err).ToNot(HaveOccurred())
+
+		// and
+		Expect(ValidatePersistedResource()).To(Succeed())
+	})
+
+	It("should apply a Mesh resource from URL", func() {
```

Comment:
How this test work without setting up a server?

I think that single test here is fine - I'd leave only `should apply a new Dataplane resource from URL`.